File: PagingJava/app/src/main/java/com/websarva/wings/android/pagingjava/data/repository/PhoneRepository.java
package com.websarva.wings.android.pagingjava.data.repository;

import androidx.paging.Pager;
import androidx.paging.PagingConfig;

import com.websarva.wings.android.pagingjava.data.local.Phone;
import com.websarva.wings.android.pagingjava.data.local.PhonePagingSource;

public class PhoneRepository {
	private static final int ITEMS_PER_PAGE = 30;

	public Pager<Integer, Phone> getAllPhoneListPager() {
		PagingConfig pagingConfig = new PagingConfig(ITEMS_PER_PAGE);
		Pager<Integer, Phone> phoneListPager = new Pager<>(pagingConfig, () -> new PhonePagingSource());
		return phoneListPager;
	}
}

File: PagingKotlin/app/src/main/java/com/websarva/wings/android/pagingkotlin/data/repository/PhoneRepository.kt
package com.websarva.wings.android.pagingkotlin.data.repository

import androidx.paging.Pager
import androidx.paging.PagingConfig
import com.websarva.wings.android.pagingkotlin.data.local.Phone
import com.websarva.wings.android.pagingkotlin.data.local.PhonePagingSource

private const val ITEMS_PER_PAGE = 50

class PhoneRepository {
	fun getAllPhoneListPager(): Pager<Int, Phone> {
		val pagingConfig = PagingConfig(ITEMS_PER_PAGE)
		val phoneListPager = Pager(pagingConfig) {PhonePagingSource()}
		return phoneListPager
	}
}
